spec/requests/categories_controller_spec.rb

```diff
@@ -372,6 +372,82 @@
         response.parsed_body["category_list"]["categories"].map { |x| x["id"] },
       ).not_to include(uncategorized.id)
     end
+
+    describe "with serialized category custom fields" do
+      class CategoryPlugin < Plugin::Instance
+        attr_accessor :enabled
+        def enabled?
+          @enabled
+        end
+      end
+
+      fab!(:plugin) { CategoryPlugin.new }
+
+      before do
+        plugin.enabled = false
+        clear_custom_fields
+      end
+
+      def add_custom_fields
+        plugin.enabled = true
+        plugin.add_to_serializer(:basic_category, :bob) { object.custom_fields["bob"] }
+        category.custom_fields["bob"] = "marley"
+        category.save
+      end
+
+      def clear_custom_fields
```

Comment:
This does not have to be a method.

Reply:
Sure: https://github.com/discourse/discourse/pull/23969/commits/19473f74dcbae14a37131f09418409f8ae4e2650